### PR TITLE
Scrum 44

### PR DESCRIPTION
**Pull request by Hsbalazs**

### CODE REVIEW

**Review by ramesz6 — APPROVED** [review]

File: backend/src/main/java/com/greenfoxacademy/backend/services/UserServiceImpl.java [hunk 1]
```diff
@@ -21,9 +22,11 @@ public class UserServiceImpl implements UserService {
...
   @Override
   public void register(RegisterUserDto userDto) {
-    User user = this.mapToEntity(userDto);
-    user.setPassword(passwordEncoder.encode(userDto.getPassword()));
-    userRepository.save(user);
+    if (emailValidation(userDto.getEmail())) {
+      User user = this.mapToEntity(userDto);
+      user.setPassword(passwordEncoder.encode(userDto.getPassword()));
+      userRepository.save(user);
+    }
```
Comment (markkovari) [inline backend/src/main/java/com/greenfoxacademy/backend/services/UserServiceImpl.java:29]: Just a question? How do we know if the user is created or not?

Comment (Hsbalazs) [reply backend/src/main/java/com/greenfoxacademy/backend/services/UserServiceImpl.java:29]: There is a method emailValidation to validate the email.
I've forgotten to code if email is already exist. Sorry....
I would put "&& !userRepository.existsByEmail(userDto.getEmail())" to the if statement to check if email already exists or not.

Comment (markkovari) [reply backend/src/main/java/com/greenfoxacademy/backend/services/UserServiceImpl.java:29]: Well you can annotate the email as unique in the database, but how can one know if the email is not valid? it is void and we are just sending 200 - Ok back? Can we somehow validate - even automatically - if it works and will work too?

If a user is not using a valid email they are not notified now.

Comment (Hsbalazs) [reply backend/src/main/java/com/greenfoxacademy/backend/services/UserServiceImpl.java:29]: I supposed that user will get feedback only on the frontend about the validation.
In backend just user with valid email will be saved if it is not exist yet.

Ok, so this case I think I need to add if statement to the register endpoint in the UserController.

**Review by carlos-ponton — APPROVED** [review]

**Pull request merged**